docs/site/migration/models/relations.md

```diff
@@ -6,8 +6,72 @@ sidebar: lb4_sidebar
 permalink: /doc/en/lb4/migration-models-relations.html
 ---
 
-{% include note.html content="
-This is a placeholder page, the task of adding content is tracked by the
-following GitHub issue:
-[loopback-next#3948](https://github.com/strongloop/loopback-next/issues/3948).
-" %}
+When you define a relation in a LoopBack 3 model JSON file, the framework will
+create the following artifacts for you automatically:
+
+- Relation metadata defining the target model, foreign key column/property, and
+  so on.
+- Repository-like methods for accessing related model instance(s), for example
+  `Category.prototype.products`.
+- An inclusion resolver to allow clients to request relation traversal in
+  queries and include related models, e.g.
+  `Product.find({include: ['category']})`
+- REST API endpoints for querying a modifying models on the other side of the
+  relation, e.g. `GET /api/categories/1/products`.
+
+In LoopBack 4, these building blocks are typically provided by the application
+developer.
+
+- Relation metadata is defined via model decorators like `@hasMany`.
+- Relation repositories implement APIs for accessing and modifying data of
+  related models.
+- Inclusion resolvers implement relation traversal in queries.
+- Relation controllers implement REST APIs for model relations.
+
+At the moment, all of these artifacts are defined via source code files which
+are typically created by running `lb4 relation`. With source code files ready to
+be edited, developers get a lot of power and flexibility in customizing the
+default behavior offered by the framework.
+
+In the future, we would like to provide declarative approach for building model
+relations, where the developer defines relation metadata and the framework
+builds all required artifacts at runtime, similarly to how LoopBack 3 works. You
+can join the discussion in the GitHub issue
+[loopback-next#2483](https://github.com/strongloop/loopback-next/issues/2483).
+
+## Migration path
+
+Follow these steps to migrate a model relation from LoopBack 3 to LoopBack 4:
+
+1. Make sure to complete all steps described in
+   [Migrating model definitions and built-in APIs](./core.md), especially
+   creation of Repository classes for models on both sides of the relation.
+
+2. Run `lb4 relation` to define the model relation in your model class, generate
```

Comment:
Maybe make it more clear like:

```
Run `lb4 relation` to create the model relations according to the relation definitions defined in lb3 model json files.
```

so that people understand where to find the relation definitions.
And we can link to the generator doc as well: https://loopback.io/doc/en/lb4/Relation-generator.html

Reply:
> according to the relation definitions defined in lb3 model json files.

Reply:
> according to the relation definitions defined in lb3 model json files.

I am concerned that this is not entirely correct. Relations can be defined from JS code too, e.g. `Product.belongsTo(Category)`.

I am not sure how to best incorporate this information into the migration guide. Let's leave it out of scope of this pull request. If you have some ideas how to improve the current text, then can you please open a follow-up pull request yourself? 

> And we can link to the generator doc as well: https://loopback.io/doc/en/lb4/Relation-generator.html

Great idea 👍